src/peakrdl_regblock/struct_generator.py

```diff
@@ -88,16 +88,31 @@ def push_struct(self, inst_name: str, array_dimensions: Optional[List[int]] = No
         self._struct_stack.append(s)
 
 
-    def add_member(self, name: str, width: int = 1, array_dimensions: Optional[List[int]] = None) -> None:
+    def add_member(
+            self,
+            name: str,
+            width: int = 1,
+            array_dimensions: Optional[List[int]] = None,
+            *,
+            lsb: int = 0,
+            signed: Optional[bool] = None,
+    ) -> None:
         if array_dimensions:
             suffix = "[" + "][".join((str(n) for n in array_dimensions)) + "]"
         else:
             suffix = ""
 
-        if width == 1:
-            m = f"logic {name}{suffix};"
+        if signed is None:
+            sign = ""
+        elif signed:
+            sign = "signed "
         else:
```

Comment:
Lets simplify this and remove the explicit `unsigned` annotation. Struct members are already declared as `logic` which is implicitly unsigned as per SV LRM 6.11.3
This also lets you collapse the type annotation of the arg to: `signed: bool = False` instead of making it `Optional[]`.